nwb_conversion_tools/datainterfaces/icephys/baseicephysinterface.py

```diff
@@ -0,0 +1,145 @@
+"""Author: Luiz Tauffer."""
+from abc import ABC
+from typing import Optional
+
+from pynwb import NWBFile
+from pynwb.device import Device
+from pynwb.icephys import IntracellularElectrode
+
+from ...basedatainterface import BaseDataInterface
+from ...tools.neo import (
+    get_number_of_electrodes,
+    get_number_of_segments,
+    write_neo_to_nwb,
+)
+from ...utils import (
+    OptionalFilePathType,
+    get_schema_from_hdmf_class,
+    get_schema_from_method_signature,
+    get_base_schema,
+)
+
+
+try:
+    from ndx_dandi_icephys import DandiIcephysMetadata
+
+    HAVE_NDX_DANDI_ICEPHYS = True
+except ImportError:
+    HAVE_NDX_DANDI_ICEPHYS = False
+
+
+class BaseIcephysInterface(BaseDataInterface, ABC):
+    """Primary class for all intracellular NeoInterfaces."""
+
+    neo_class = None
+
+    @classmethod
+    def get_source_schema(cls):
+        source_schema = get_schema_from_method_signature(class_method=cls.__init__, exclude=[])
+        return source_schema
+
+    def __init__(self, file_paths: list):
+        super().__init__(file_paths=file_paths)
+
+        self.readers_list = list()
+        for f in file_paths:
+            self.readers_list.append(self.neo_class(filename=f))
+
+        self.subset_channels = None
+        self.n_segments = get_number_of_segments(neo_reader=self.readers_list[0], block=0)
+        self.n_channels = get_number_of_electrodes(neo_reader=self.readers_list[0])
```

Comment:
Looking ahead towards other potential icephys interfaces; how common is it to have multiple files? Is it universal? Or just something specific to ABF?

In general, the base interface for such things would be expected to take things dynamically and leave format-specific file/folder structures to the child interfaces.

Reply:
I don't know actually, I'd say the best thing is to wait until we have another icephys file type to be covered by this base class and see